src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IAsyncEnumerable.cs

```diff
@@ -8,6 +8,7 @@ namespace System.Collections.Generic
     /// <summary>Exposes an enumerator that provides asynchronous iteration over values of a specified type.</summary>
     /// <typeparam name="T">The type of values to enumerate.</typeparam>
     public interface IAsyncEnumerable<out T>
+        where T : allows ref struct
```

Comment:
@AlekseyTs, @cston did we include the possibility of the iterator in `await foreach` being a `ref struct` in our tests?

Reply:
@jaredpar 

> did we include the possibility of the iterator in await foreach being a ref struct in our tests?

There is a section about `await foreach` in the speclet. We have corresponding tests. If you have something else in mind, you have to be more specific about what scenario you are referring to.

Reply:
Thats' what I wanted to hear. Thanks!